ftl/python/layer_builder.py

```diff
@@ -239,8 +239,12 @@ def __init__(self,
         self._pkg_descriptor = pkg_descriptor
 
     def GetCacheKeyRaw(self):
-        return "%s %s %s" % (self._pkg_descriptor[0], self._pkg_descriptor[1],
-                             self._dep_img_lyr.GetCacheKeyRaw())
+        if self._dep_img_lyr:
```

Comment:
ah, this might require more thinking then.  You can't just remove self._dep_img_lyr.GetCacheKeyRaw() from the cache key as all of the get's for the cache key will use this value so it'll never hit.  You'll need to use the old cache key and supply the _dep_img_lyr or remove the concept altogether.  Even in the case that python packages are the same for each python version, the path is different.  That is what that piece of the cache key encapsulates

Reply:
ah yeah I see now. that's definitely something we want to leave in there, I realized that I can create an `InterpreterLayerBuilder` with the defaults and use that as the `dep_img_lyr`. this way the cache key will at least represent that we used the default paths to install the package.